crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2389,28 +2414,48 @@ impl<'db> TypeInferenceBuilder<'db> {
             )
         };
 
-        let unbound_ty = if may_be_unbound {
-            Some(self.lookup_name(name))
-        } else {
-            None
-        };
+        let mut definitions = definitions.peekable();
+        let has_definitions = definitions.peek().is_some();
```

Comment:
This seems unnecessary/redundant if we will always call `bindings_ty`, because the result of calling `bindings_ty` will also tell us if the name has bindings or not.